src/plugins/intel_npu/tests/functional/behavior/npuw/memory.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+#include <functional>
+#include <chrono>
+#include <fstream>
+#include <algorithm>
+#include "openvino/runtime/tensor.hpp"
+
+namespace {
+
+float ms_to_run(std::function<void()> &&body) {
+    namespace khr = std::chrono;
+    const auto s = khr::steady_clock::now();
+    body();
+    const auto f = khr::steady_clock::now();
+    return khr::duration_cast<khr::microseconds>(f - s).count() / 1000.0;
+};
+
+struct MSec {
+    constexpr static const char *name = "ms";
+};
+
+template<typename T, typename U>
+class metric {
+    std::vector<T> records;
+    T vmin = std::numeric_limits<T>::max();
+    T vmax = std::numeric_limits<T>::min();
+    std::string unit;
+
+public:
+    void operator += (T &&t) {
+        vmin = std::min(vmin, t);
+        vmax = std::max(vmax, t);
+        records.push_back(std::move(t));
+    }
+    float avg() const {
+        float acc = 0.f;
+        for (auto &&t : records) {
+            acc += static_cast<float>(t) / records.size();
+        }
+        return acc;
+    }
+    T med() const {
+        std::vector<T> cpy(records);
+        std::nth_element(cpy.begin(), cpy.begin() + cpy.size()/2, cpy.end());
```

Comment:
interesting why do we need intermediate vector? just use c.begin() c.end() in nth_element call

Reply:
Removed that file